lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,562 @@
+{
+  GList *result = NULL, *p;
+
+  g_return_val_if_fail (GST_IS_PLAYER_MEDIA_INFO (info), NULL);
+
+  for (p = info->stream_list; p != NULL; p = p->next) {
+    GstPlayerStreamInfo  *sinfo = (GstPlayerStreamInfo*) p->data;
+
+    if (GST_IS_PLAYER_SUBTITLE_INFO (sinfo)) {
+      GstPlayerSubtitleInfo  *tinfo = (GstPlayerSubtitleInfo*) sinfo;
+
+      result = g_list_append (result, tinfo);
+    }
+  }
+
+  return result;
+}
+
+/**
+ * gst_player_audio_info_get_streams:
+ * @info: a #GstPlayerMediaInfo
+ *
+ * Returns: (transfer full) (element-type GstPlayerAudioInfo): A #GList of
+ * matching #GstPlayerAudioInfo. The caller should free it with
+ * gst_player_stream_info_list_free().
+ */
+GList*
+gst_player_media_info_get_audio_streams (const GstPlayerMediaInfo *info)
+{
+  GList *result = NULL, *p;
+
+  g_return_val_if_fail (GST_IS_PLAYER_MEDIA_INFO (info), NULL);
+
+  for (p = info->stream_list; p != NULL; p = p->next) {
+    GstPlayerStreamInfo  *sinfo = (GstPlayerStreamInfo*) p->data;
+
+    if (GST_IS_PLAYER_AUDIO_INFO (sinfo)) {
+      GstPlayerAudioInfo  *ainfo = (GstPlayerAudioInfo*) sinfo;
+
+      result = g_list_append (result, ainfo);
+    }
+  }
+
+  return result;
+}
+
+/**
+ * gst_player_audio_info_get_language:
+ * @info: a #GstPlayerAudioInfo
+ *
+ * Returns: the language of the stream, or NULL if unknown.
+ */
+const gchar*
+gst_player_audio_info_get_language(const GstPlayerAudioInfo* info)
+{
+  gchar *language = NULL;
+
+  g_return_val_if_fail (GST_IS_PLAYER_MEDIA_INFO (info), NULL);
+
+  if (info->language)
+    language = g_strdup (info->language);
+
+ return language;
+}
+
+/**
+ * gst_player_media_stream_info_list_free:
+ * @infos: a #GList of #GstPlayerStreamInfo
+ *
+ * frees the #GList.
+ */
+void
+gst_player_media_info_stream_info_list_free (GList *infos)
+{
+  g_list_free (infos);
+}
+
+/* Video information */
+G_DEFINE_TYPE (GstPlayerVideoInfo, gst_player_video_info,
+    GST_TYPE_PLAYER_MEDIA_INFO);
+
+static void
+gst_player_video_info_init (GstPlayerVideoInfo * info)
+{
+  /* Nothing to do here */
+}
+
+/**
+ * gst_player_video_info_get_width:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the width of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_width (const GstPlayerVideoInfo *info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->width;
+}
+
+/**
+ * gst_player_video_info_get_height:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the height of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_height (const GstPlayerVideoInfo *info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->height;
+}
+
+/**
+ * gst_player_video_info_get_framerate_num:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the framerate numerator of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_framerate_num (const GstPlayerVideoInfo *info)
```

Comment:
void gst_player_video_info_get_framerate (const GstPlayerVideoInfo *info, guint *fps_n, guint *fps_d) would be better, and then making these out-parameters